### ISSUE TITLE
`--dry-run` option is not working

### ISSUE THREAD

**Issue by mwestphal** [f3d-app/f3d#1637]
**Describe the bug**
F3D usually uses a config file, but there is an option to disable this behavior, `--dry-run`. It seems to be non-functionnal currently

**To Reproduce**
Steps to reproduce the behavior, using a built f3d with BUILD_TESTING enabled:
 1. Open the file using `f3d --config=config_build --dry-run --verbose
 2. F3D appears with axis visible, it should not
 3. Check the output, a config file is read

**Expected behavior**
Config file should not be read

**Additional context**
in `F3DStarter.cxx:655`, although the comment says that dry-run is being checked, its actually not. Add the check here by checking the cli option and position the boolean accordingly. Also add a test for it in `application/testing/CMakeLists.txt`

**Comment by snoyer** [comment]
Would this be the right time to rename this argument to `--no-config` or something?
`--dry-run` is [used in many CLI applications](https://www.google.com/search?q=%22--dry-run%22+site%3Alinux.die.net) to mean "run as normal but don't actually write/change anything" so it's a bit confusing (F3D's `--no-render` is closer to what `--dry-run` usually means)

**Comment by mwestphal** [comment]
Good point @snoyer

**Comment by mwestphal** [comment]
It is open!

**Comment by mwestphal** [comment]
Hum, since you are working on another issue, I'd like to keep this one open for other developpers with no experience on the project if thats ok

**Comment by Yogesh9000** [comment]
sure thats fine

**Comment by rehanganapathy** [comment]
Hey there, this looks interesting, if no one else is working on it, can i pick it up? Thank you so much!
@mwestphal

**Comment by mwestphal** [comment]
Sounds good! @rehanganapathy , go for it :)

**Comment by mwestphal** [comment]
@rehanganapathy any news on this ?

**Comment by mwestphal** [comment]
no feedback, unassigning it

**Comment by t-h2o** [comment]
Hello!

With this patch, the config value is never set.

```patch
 application/F3DStarter.cxx | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/application/F3DStarter.cxx b/application/F3DStarter.cxx
index e1cc563a..6b00c5bc 100644
--- a/application/F3DStarter.cxx
+++ b/application/F3DStarter.cxx
@@ -713,12 +713,16 @@ int F3DStarter::Start(int argc, char** argv)
   // but this duplicate the initialization value as it is present in
   // F3DOptionTools::DefaultAppOptions too
   bool dryRun = false;
+  if (cliOptionsDict.find("dry-run") != cliOptionsDict.end())
+  {
+    dryRun = f3d::options::parse<bool>(cliOptionsDict["dry-run"]);
+  }
   if (cliOptionsDict.find("no-render") != cliOptionsDict.end())
   {
     dryRun = f3d::options::parse<bool>(cliOptionsDict["no-render"]);
   }
   std::string config;
-  if (cliOptionsDict.find("config") != cliOptionsDict.end())
+  if (!dryRun && cliOptionsDict.find("config") != cliOptionsDict.end())
   {
     config = f3d::options::parse<std::string>(cliOptionsDict["config"]);
   }
```

Is it correct and/or a good starting point?

**Comment by mwestphal** [comment]
It is definitely a good starting point! Do you want to open a PR with this patch @t-h2o ?

**Comment by t-h2o** [comment]
> It is definitely a good starting point! Do you want to open a PR with this patch @t-h2o ?

Done: https://github.com/f3d-app/f3d/pull/1720

**Comment by void-inject** [comment]
That's me, vincenzo. i am getting this issue

**Comment by mwestphal** [comment]
Sorry, this issue is not available

**Comment by mwestphal** [comment]
FIxed by #1720
